### ISSUE TITLE
Include Empty Rows

### ISSUE THREAD

**Issue by harshitgupta** [DiegoZoracKy/convert-excel-to-json#11]
Hi, 

First of all thank you for the module. Its one of the rare modules that 

- support both xls and xlsx files
- gives the data cell by cell

One problem I found is that it ignores empty lines by default. My app's business logic requires me to pick up rows based upon user-given index values (which include empty lines).

It would be great if you could make it config based rather the current approach. 

I looked into the code and found I need to comment following lines in order enable empty lines:

```
           // if (columnData === '') {
           //     continue;
           // }
```

```
        // Cleaning empty
        // rows = rows.filter(v => v !== null && v !== undefined);
```

Can you please confirm that I am not missing any other lines?

Thanks.

**Comment by DiegoZoracKy** [comment]
You are right. It does that by default. What do you think about a config to indicate when it should return the empty values (being the current behavior of not returning the empty the default value)? 

As you are already on the right snippet of the code, would you like to implement it?

**Comment by DiegoZoracKy** [comment]
Hi @harshitgupta, 

Are you still interested in working on that? No pressure, just checking to see if we should keep this issue open.

**Comment by NunoChumbo** [comment]
Hi, i too need for the column data even if cell is empty, but as for the empty lines it may stayas it is.
Can you please apply the suggested fix. 
Thank you

**Comment by harshitgupta** [comment]
sorry for the delay but I just created the pull request. 

Please let me know if you have any questions.

Thanks @NunoChumbo for the reminder.

**Comment by DiegoZoracKy** [comment]
Hi @ harshitgupta,

Thanks for creating the PR. It will handle very well this case. I just wrote one comment there that I would like to see your answer before merging it.

**Comment by harshitgupta** [comment]
@DiegoZoracKy Sure but I don't see any comment there. 

<img width="1005" alt="screen shot 2018-11-04 at 10 28 14 pm" src="https://user-images.githubusercontent.com/1644577/47978064-2c186780-e081-11e8-988c-235c80f21199.png">

Could you please point me to it?

**Comment by DiegoZoracKy** [comment]
Sure, it is on the changes section:

https://github.com/DiegoZoracKy/convert-excel-to-json/pull/20/files

**Comment by harshitgupta** [comment]
well, it's embarrassing. I still don't see it, neither did i get any notification for it. do you have to assign it to me or something?

**Comment by DiegoZoracKy** [comment]
That's weird. Anyway, here is the screenshot of it:

![img_20181105_025235](https://user-images.githubusercontent.com/2091071/47978619-00f43f00-e0a6-11e8-8c6b-1451c3ccd5c2.jpg)

**Comment by harshitgupta** [comment]
On my side its looks like this:

<img width="1011" alt="screen shot 2018-11-04 at 10 56 43 pm" src="https://user-images.githubusercontent.com/1644577/47978726-09884d80-e085-11e8-87ee-ad5f8f07b8a1.png">

no sign of any comment or anything.

But anyways to answer your question I would say that since we are dealing with array which starts with index of 0 and the `row` at line # 49 always start from `1` (assuming first line is not empty) as first cell will always be `A1` in the same case. That means nothing ever will get assign to `row[0]`, thus it will always be `undefined`. 

Second it has never been a problem so far because in the existing using the following code you have been getting rid of rows that are null or undefined.

`rows = rows.filter(v => v !== null && v !== undefined);`

Please let me know if it makes sense?

**Comment by DiegoZoracKy** [comment]
Thanks for your answer and your PR @harshitgupta. It is already merged and published on npm (v 1.5.0)

@NunoChumbo might want to take a look.

**Comment by DiegoZoracKy** [comment]
PR merged: https://github.com/DiegoZoracKy/convert-excel-to-json/pull/20
